scipy/optimize/_basinhopping.py

```diff
@@ -645,25 +643,32 @@ def basinhopping(func, x0, niter=100, T=1.0, stepsize=0.5,
     at minimum -0.7425 accepted 1
     at minimum -0.9073 accepted 1
 
-
     The minimum at -1.0109 is actually the global minimum, found already on the
     8th iteration.
 
-    Now let's implement bounds on the problem using a custom ``accept_test``:
+    Now let's restrict new accepted solutions to be within some
+    bounds by providing a custom `accept_test`:
 
-    >>> class MyBounds:
-    ...     def __init__(self, xmax=[1.1,1.1], xmin=[-1.1,-1.1] ):
-    ...         self.xmax = np.array(xmax)
-    ...         self.xmin = np.array(xmin)
-    ...     def __call__(self, **kwargs):
-    ...         x = kwargs["x_new"]
-    ...         tmax = bool(np.all(x <= self.xmax))
-    ...         tmin = bool(np.all(x >= self.xmin))
+    >>> class MyAcceptanceBounds:
+    ...     def __init__(self):
+    ...         self.xmax = np.array([1.1, 1.1])
+    ...         self.xmin = np.array([-1.1, -1.1])
+    ...     def __call__(self, f_new, x_new, f_old, x_old):
+    ...         tmax = bool(np.all(x_new <= self.xmax))
+    ...         tmin = bool(np.all(x_new >= self.xmin))
     ...         return tmax and tmin
 
-    >>> mybounds = MyBounds()
+    >>> accept_bounds = MyAcceptanceBounds()
     >>> ret = basinhopping(func2d, x0, minimizer_kwargs=minimizer_kwargs,
-    ...                    niter=10, accept_test=mybounds)
+    ...                    niter=10, accept_test=accept_bounds)
+
+    .. note::
+
+        This snippet only restrict new candidates to be within bounds.
+        The local minimization and stepping process can still evaluate points
+        outside bounds. To completely restrict the problem to specific bounds,
+        bounds need to be defined for all `minimizer_kwargs`, `take_step` and
+        `accept_test`.
```

Comment:
I think that if both the minimizer and take_step respect bounds then the candidate evaluated by accept_test will always be within bounds.  So it would not be strictly necessary for accept_test to know about the bounds.